### PR TITLE
Fix #158, Implement exception storage in PSP

### LINKED ISSUE #158: Implement exception logging/capture in PSP
**Is your feature request related to a problem? Please describe.**
Exception handling in CFE is not possible due to race conditions and differences between platforms, so the PSP needs to take a greater role in handling these events.

**Describe the solution you'd like**
PSP should capture exception contexts locally, and provide a more abstract API for CFE to poll and retrieve them rather than pushing direct.

**Additional context**
Related to nasa/cfe#411, nasa/cfe#76

**Requester Info**
Joseph Hickey, Vantage Systems, Inc.

### PR DESCRIPTION
**Pull request by jphickey**
**Describe the contribution**
Implements exception capture and storage within the PSP.
Provides a new API so that CFE can poll for exceptions at a later time and also record the information in the ER log.

**Testing performed**
Add a divide by zero "bug" to sample_app
Confirm that the PSP captures the exception context correctly and it is later logged by CFE
Confirm SIGINT, SIGQUIT, and SIGTERM are appropriately handled on POSIX

**Expected behavior changes**
Exception handling is now implemented on POSIX.  The initial config will capture arithmetic errors (SIGFPE) as well as SIGINT and SIGTERM.  The latter is notable in that there is no longer a separate handler for SIGINT - it is now treated as an exception and goes through the normal process which ends up "restarting" CFE, which on pc-linux causes the process to exit normally.  

Importantly, as CTRL+C is frequently used during testing, this means that there is now a mechanism through which the exception code will get used during normal test cycles.  Otherwise the code is very infrequently used and could suffer from bit-rot.

**System(s) tested on**
Ubuntu 20.04 LTS 
VxWorks 6.9 on MCP750
i686-rtems4.11 (pc-rtems) on  QEMU (no exception handling here, just tested to ensure nothing broke)

**Additional context**
Goes along with nasa/cfe#411, nasa/cfe#76

**Contributor Info - All information REQUIRED for consideration of pull request**
Joseph Hickey, Vantage Systems, Inc.

### CODE REVIEW

**Comment by jphickey** [comment]
All components of this are now rebased.  Ready for merge.
This was reviewed in a special topic on 2020-04-30.

**Comment by skliper** [comment]
Need to resolve CI failures

File: fsw/mcp750-vxworks/src/cfe_psp_memory.c [hunk 1]
```diff
@@ -337,19 +347,96 @@ int32 CFE_PSP_InitProcessorReservedMemory( uint32 RestartType )
...
+    ** of the memory space.
+    */
+    start_addr = (cpuaddr) sysMemTop();
+    end_addr = start_addr;
+
+    memset(&CFE_PSP_ReservedMemoryMap, 0, sizeof(CFE_PSP_ReservedMemoryMap));
+
+    CFE_PSP_ReservedMemoryMap.BootPtr = (CFE_PSP_ReservedMemoryBootRecord_t *)end_addr;
+    end_addr += sizeof(CFE_PSP_ReservedMemoryBootRecord_t);
+    end_addr = (end_addr + CFE_PSP_MEMALIGN_MASK) & ~CFE_PSP_MEMALIGN_MASK;
```
Comment (skliper) [inline fsw/mcp750-vxworks/src/cfe_psp_memory.c:394]: Did you look into the routines Doyle suggested?

**Comment by astrogeco** [comment]
CCB 20200506 - Had special topic on April 30 2020. @acudmore will review later this week.

**Pull request merged**